scipy/optimize/_differentialevolution.py

```diff
@@ -106,17 +106,24 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
         If True (default), then `scipy.optimize.minimize` with the `L-BFGS-B`
         method is used to polish the best population member at the end, which
         can improve the minimization slightly.
-    init : string, optional
-        Specify how the population initialization is performed. Should be
+    init : str or array-like, optional
+        Specify which type of population initialization is performed. Should be
         one of:
 
             - 'latinhypercube'
             - 'random'
+            - array specifying subset of the initial population. The array can
+              be of shape ``(len(x),)`` or ``(M, len(x))``, where len(x) is the
+              number of parameters. `init` is clipped to `bounds` before use.
 
         The default is 'latinhypercube'. Latin Hypercube sampling tries to
-        maximize coverage of the available parameter space. 'random' initializes
-        the population randomly - this has the drawback that clustering can
-        occur, preventing the whole of parameter space being covered.
+        maximize coverage of the available parameter space. 'random'
+        initializes the population randomly - this has the drawback that
+        clustering can occur, preventing the whole of parameter space being
+        covered. Use of an array to specify a population subset could be used,
+        for example, to create a tight bunch of initial guesses in an location
+        where the solution is known to exist, thereby reducing time for
+        convergence.
```

Comment:
This "subset" approach is not explained and quite unusual. What happens is that ``latinhypercube`` is still used, but an arbitrary part of the initial population that that generates is then *overwritten* with the provided array. Why not just take the given array as the full initial population.

Note that this also doesn't accomplish the goal stated in the PR of letting users try their own initialization method, it'll be a weird mix of two methods.

Reply:
I was making the changes for how I would use it. Here are the situations I originally envisaged:

1) You want to initialise the population with e.g. latin hypersquares, but you also have an idea for what the best fit should be (c.f. the initial guess). So to seed the minimization you supply an array of shape `(len(x),)`.
2) You want to seed the population with various initial guesses (e.g. reflecting a covariance matrix)
3) You want to specify the entire population.

In order to achieve 1) you have to have an array to copy into, so that's how I came up with the current design.

Perhaps the following could be used instead:

1) if the initial guess has shape `(len(x),)`, i.e. 1-D, then initialise with latin hypersquares, but only set the initial population member.
2) if the initial guess is 2D, then it's assumed to specify the entire population.